lib/airbrake/rack/instrumentable.rb

```diff
@@ -15,12 +17,99 @@ module Rack
     # @since v9.2.0
     module Instrumentable
       def airbrake_capture_timing(method_name, label: method_name.to_s)
-        alias_method "#{method_name}_without_airbrake", method_name
+        mod = ::Airbrake::Rack::Instrumentable
+        if mod.should_prepend?(self, method_name)
+          mod.prepend_airbrake_capture_timing(self, method_name, label)
+        else
+          mod.chain_airbrake_capture_timing(self, method_name, label)
+        end
+        method_name
+      end
+
+      # @api private
+      def __airbrake_capture_timing_module__
```

Comment:
Even though this is marked as private API, the method is still public and therefore extenders of the module will have access to it (for no good reason). Perhaps you left it private for easier testability but I would prefer not to leak implementation details to user space. We can work around this in the tests by mocks or calling `__send__`.

Reply:
We're leaking an instance variable in any case. The method only exists at all for testing purposes, and in case we have to do something weird for thread safety in the future, though I made sure that it would be thread-safe enough for our purposes by saving the result in a local variable at the call site.

There is no harm in making the method available; it's only api private because it's useless to the user, so there's no point in documenting it.

Reply:
Sorry, I tried to remove the method and just use the instance variable, and I couldn't make it thread-safe and still work without a lot of effort. I'll make the method private so it doesn't add a public method in controllers.